src/adapters/openx.js

```diff
@@ -1,130 +1,228 @@
+        beaconParams.ts = adUnit.ts;
+        addBidResponse(adUnit, bid);
+      } 
+      buildBoPixel(adUnit.creative[0], beaconParams);
+    }
+  };
+
+  function getViewportDimensions(isIfr) {
+    var width,
+      height,
+      tWin = window,
+      tDoc = document,
+      docEl = tDoc.documentElement,
+      body;
+
+    if (isIfr) {
+      tWin = window.top;
+      tDoc = window.top.document;
+      docEl = tDoc.documentElement;
+      body = tDoc.body;
+
+      width = tWin.innerWidth || docEl.clientWidth || body.clientWidth;
+      height = tWin.innerHeight || docEl.clientHeight || body.clientHeight;
+    } else {
+      docEl = tDoc.documentElement;
+      width = tWin.innerWidth || docEl.clientWidth;
+      height = tWin.innerHeight || docEl.clientHeight;
     }
 
-    _requestBids();
+    return `${width}x${height}`;
   }
 
-  function _requestBids() {
+  function makePDCall(pixelsUrl) {
+    let pdFrame = utils.createInvisibleIframe();
+    let name = 'openx-pd';
+    pdFrame.setAttribute("id", name);
+    pdFrame.setAttribute("name", name);
+    let rootNode = document.body;
 
-    if (scriptUrl) {
-      adloader.loadScript(scriptUrl, function () {
-        var i;
-        var POX = OX();
+    if (!rootNode) {
+      return;
+    }
 
-        if (opts.pageURL) {
-          POX.setPageURL(opts.pageURL);
-        }
+    pdFrame.src = pixelsUrl;
 
-        if (opts.refererURL) {
-          POX.setRefererURL(opts.refererURL);
+    if (pdNode) {
+      pdNode.parentNode.replaceChild(pdFrame, pdNode);
+      pdNode = pdFrame;
+    } else {
+      pdNode = rootNode.appendChild(pdFrame);
+    }
+  }
+
+  function addBidResponse(adUnit, bid) {
+    var bidResponse = bidfactory.createBid(adUnit ? CONSTANTS.STATUS.GOOD : CONSTANTS.STATUS.NO_BID);
+    bidResponse.bidderCode = BIDDER_CODE;
+
+    if (adUnit) {
+      let creative = adUnit.creative[0];
+      bidResponse.ad = adUnit.html;
+      bidResponse.cpm = Number(adUnit.pub_rev) / 1000;
+      bidResponse.ad_id = adUnit.adid;
+      if (adUnit.deal_id) {
+        bidResponse.dealId = adUnit.deal_id;
+      }
+      if (creative) {
+        bidResponse.width = creative.width;
+        bidResponse.height = creative.height;
+      }
+    }
+    bidmanager.addBidResponse(bid.placementCode, bidResponse);
+  }
+
+  function buildQueryStringFromParams(params) {
+    for (var key in params) {
+      if (params.hasOwnProperty(key)) {
+        if (!params[key]) {
+          delete params[key];
         }
+      }
+    }
+    return utils._map(Object.keys(params), key => `${key}=${params[key]}`)
+      .join('&');
+  }
+
+  function buildBoPixel(creative, params) {
+    let img = new Image();
+    let recordPixel = creative.tracking.impression;
+    let boBase = recordPixel.match(/([^?]+\/)ri\?/);
 
-        if (opts.pgid) {
-          POX.addPage(opts.pgid);
+    if (boBase) {
+      img.src = `${boBase[1]}bo?${buildQueryStringFromParams(params)}`;
+      return img;
```

Comment:
You're right, the return is not needed. This might be a relic from testing phases, I will remove the return.